packages/api/src/hooks/useP2PAdvertCreate.ts

```diff
@@ -0,0 +1,39 @@
+import { useCallback } from 'react';
+import useMutation from '../useMutation';
+// import useInvalidateQuery from '../useInvalidateQuery';
+
+type TPayload = Parameters<ReturnType<typeof useMutation<'p2p_advert_create'>>['mutate']>[0]['payload'];
+
+/** A custom hook that creates a P2P advert. This can only be used by an approved P2P advertiser.
+ * 
+ * To create an advert, specify the following payload arguments in the `mutate` call:
+ * @example
+ *  mutate({
+        description: 'Please transfer to account number 1234',
+        type: 'buy',
+        amount: 100,
+        max_order_amount: 50,
+        min_order_amount: 20,
+        payment_method: 'bank_transfer',
+        rate: 4.25,
+    });
+ * 
+*/
+const useP2PAdvertCreate = () => {
+    // const invalidate = useInvalidateQuery();
+    const { mutate: _mutate, ...rest } = useMutation('p2p_advert_create', {
+        onSuccess: () => {
+            // TODO: Invalidate P2P advert list hook once the hook is implemented
```

Comment:
Why dont we do P2P advert list hook first then do dis

Reply:
yeaa I'll do the p2pAdvertList hook first, cuz this one's dependent onit 🤣

Reply:
We can do a new thing here. `onSuccess` append the response from p2p advert create to p2p advert list cache. And `onSettled` we invalidate p2p advert list 🔥

Reply:
onSettled is when mutation is done (either success or not)

Reply:
Alright, but in the documentation, I noticed that it's marked as deprecated, and I couldn't locate the alternative method for it.

Reply:
It could append this newly created advert other `advertList` queries with filters in it, so I think its not best to append it to the cache, if the queries with filter (such as only `buy` adverts) is appended and the newly created advert is a `sell` advert, it could cause a visual bug where this newly created advert appears in the `buy` list of adverts 😓